src/shared/tasks/sentryconfig/tasksentryconfig.cpp

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "tasksentryconfig.h"
+
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonValue>
+#include <QStringList>
+
+#include "appconstants.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include "networkrequest.h"
+#include "sentry/sentryadapter.h"
+#include "settingsholder.h"
+
+namespace {
+Logger logger("TaskSentryConfig");
+}
+
+TaskSentryConfig::TaskSentryConfig() : Task("TaskSentryConfig") {
+  MZ_COUNT_CTOR(TaskSentryConfig);
+}
+
+TaskSentryConfig::~TaskSentryConfig() { MZ_COUNT_DTOR(TaskSentryConfig); }
+
+void TaskSentryConfig::run() {
+  NetworkRequest* request = new NetworkRequest(this, 200);
+
+  QUrl url(AppConstants::apiBaseUrl());
+  url.setPath("/api/v1/vpn/crashreporting");
+
+  connect(request, &NetworkRequest::requestFailed, this,
+          [this](QNetworkReply::NetworkError error, const QByteArray&) {
+            Q_UNUSED(error);
+            logger.error() << "Failed to get Sentry info";
+            emit completed();
+          });
+  connect(request, &NetworkRequest::requestCompleted, this,
+          [this](const QByteArray& data) {
+            if (handleNetworkResponse(data)) {
+              // We now should have sentry info - let's init
+              SentryAdapter::instance()->init();
```

Comment:
Consider moving this into handleNetworkResponse for better cohesion, instead of depending on the side effect that handleNetworkResponse has set the sentry config info into SettingsHolder.